controllers/storagecluster/odfinfoconfig.go

```diff
@@ -0,0 +1,248 @@
+package storagecluster
+
+import (
+	"fmt"
+	operatorsv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	ocsv1 "github.com/red-hat-storage/ocs-operator/api/v4/v1"
+	"github.com/red-hat-storage/ocs-operator/v4/controllers/util"
+	rookCephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/yaml"
+	"strings"
+	"sync"
+)
+
+type ConnectedClient struct {
+	NamespacedName types.NamespacedName `yaml:"metadata"`
+	ClusterID      string               `yaml:"clusterId"`
+}
+type InfoStorageCluster struct {
+	NamespacedName          types.NamespacedName `yaml:"metadata"`
+	StorageProviderEndpoint string               `yaml:"storageProviderEndpoint"`
+	CephClusterFSID         string               `yaml:"cephClusterFSID"`
+}
+
+type OdfInfoData struct {
+	OdfVersion          string             `yaml:"odfVersion"`
+	OdfDeploymentType   string             `yaml:"odfDeploymentType"`
+	Clients             []ConnectedClient  `yaml:"clients"`
+	StorageCluster      InfoStorageCluster `yaml:"storageCluster"`
+	StorageClusterCount int                `yaml:"storageClusterCount"`
+	StorageSystemName   string             `yaml:"storageSystemName"`
+	IsDROptimized       bool               `yaml:"isDROptimized"`
+}
+
+const (
+	OdfInfoKeyName            = "config.yaml"
+	OdfDeploymentTypeExternal = "external"
+	OdfDeploymentTypeInternal = "internal"
+	RookCephMonSecretName     = "rook-ceph-mon"
+	FsidKey                   = "fsid"
+	OdfOperatorNamePrefix     = "odf-operator"
+	OdfInfoConfigMapName      = "odf-info"
+	OdfInfoMapKind            = "ConfigMap"
+)
+
+type odfInfoConfig struct{}
+
+var mutex sync.RWMutex
+
+// ensureCreated ensures that a ConfigMap resource exists with its Spec in
+// the desired state.
+func (obj *odfInfoConfig) ensureCreated(r *StorageClusterReconciler, sc *ocsv1.StorageCluster) (reconcile.Result, error) {
+	operatorNamespace, err := util.GetOperatorNamespace()
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	odfInfoConfigMap := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      OdfInfoConfigMapName,
+			Namespace: operatorNamespace,
+		},
+	}
```

Comment:
I don't understand what difference would that make. Care to provide more input?

Reply:
this was my reasoning, it reduces an explicit import (metav1), keeps everything to the left side (vs nested), we'll not be setting more fields and only these two.

Reply:
These two _plus_ `Data`. I think it follows the convention of the code base.

Reply:
Sry, is there a question for me?

Reply:
Data shouldn't be initialized outside of `CreateOrUpdate` it will affect the result of the `Get` operation inside the  `CreateOrUpdate` and will treat it more like a shallow merge. Preferably we want the object at the start of the mutator function to be identical to how it looks like in ETCD. This is important because it affects the `deepEqual` operation at the end of `CreateOrUpdate`